state/pgstatestorage/l1infotree.go

```diff
@@ -49,3 +45,18 @@ func (p *PostgresStorage) GetAllL1InfoRootEntries(ctx context.Context, dbTx pgx.
 	}
 	return entries, nil
 }
+
+func (p *PostgresStorage) GetLatestIndex(ctx context.Context, dbTx pgx.Tx) (uint, error) {
+	const getLatestIndexSQL = `SELECT max(l1_info_tree_index) as l1_info_tree_index FROM state.exit_root 
+		WHERE l1_info_tree_index IS NOT NULL`
+	var l1InfoTreeIndex *uint
+	e := p.getExecQuerier(dbTx)
+	err := e.QueryRow(ctx, getLatestIndexSQL).Scan(&l1InfoTreeIndex)
+	if err != nil {
+		return 0, err
+	}
+	if l1InfoTreeIndex == nil {
+		return 0, state.ErrNotFound
+	}
```

Comment:
In case of no rows on DB, maybe is better to return a 0 , nil

Reply:
If returns 0, then the l1InfoTree will fail because index 0 is used as the first element. So it needs to detect the initial situation to compute the tree successfully